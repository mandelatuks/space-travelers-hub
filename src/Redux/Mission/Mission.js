import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

const url = 'https://api.spacexdata.com/v3/missions';

export const FetchMissions = createAsyncThunk(
  'missions/fetchMissions',
  async () => {
    const response = await fetch(url);
    const info = await response.json();
    const missionArray = [];
    info.map((e) => {
      const mission = {
        mission_id: e.mission_id,
        mission_name: e.mission_name,
        description: e.description,
        reserved: false,
      };
      missionArray.push(mission);
      return missionArray;
    });
    return missionArray;
  },
);

export const missionsSlice = createSlice({
  name: 'missions',
  initialState: [],
  reducers: {
    updateMission: (state, action) => state.map(
      (mission) => {
        if (mission.mission_id === action.payload) {
          return { ...mission, reserved: !mission.reserved };
        }
        return mission;
      },
    ),
  },
  extraReducers: (builder) => {
    builder
      .addCase(FetchMissions.fulfilled, (state, action) => action.payload);
  },
});
export const { updateMission } = missionsSlice.actions;
export const selectMission = (state) => state.missions;
export default missionsSlice.reducer;
