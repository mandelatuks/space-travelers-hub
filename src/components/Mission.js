import React from 'react';
import PropTypes from 'prop-types';

class Mission extends React.PureComponent {
  render() {
    const { missions } = this.props;
    return (
      <section className="missiontable">
        <table>
          <thead>
            <tr>
              <th><h4>Mission</h4></th>
              <th><h4>Description</h4></th>
              <th><h4>Status</h4></th>
              <th><h4>.</h4></th>
            </tr>
          </thead>
          <tbody>
            {missions.map((mission) => (
              <tr key={mission.mission_id}>
                <td><h4>{mission.mission_name}</h4></td>
                <td><h4>{mission.description}</h4></td>
                <td><h4 className="non-member">Not a Member</h4></td>
                <td><button type="button" className="cancel-btn">Join Mission</button></td>
              </tr>
            ))}
          </tbody>
        </table>
      </section>
    );
  }
}

Mission.propTypes = {
  missions: PropTypes.string.isRequired,
};

export default Mission;
