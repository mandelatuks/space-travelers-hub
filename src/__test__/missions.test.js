import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/Configurestore';
import Mission from '../components/pages/MissionPage';

test('test the MissionPage snapshot', () => {
  const missionRenderer = renderer.create(
    <Provider store={store}>
      <BrowserRouter>

        <Mission />
      </BrowserRouter>
    </Provider>
    ,
  ).toJSON();
  expect(missionRenderer).toMatchSnapshot();
});
