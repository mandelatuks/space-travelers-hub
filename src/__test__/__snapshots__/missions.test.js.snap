// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the MissionPage snapshot 1`] = `
<section
  className="missiontable"
>
  <table>
    <thead>
      <tr>
        <th>
          <h4>
            Mission
          </h4>
        </th>
        <th>
          <h4>
            Description
          </h4>
        </th>
        <th>
          <h4>
            Status
          </h4>
        </th>
        <th>
          <h4>
            .
          </h4>
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</section>
`;
